src/schema/v2/__tests__/fair.test.js

```diff
@@ -301,22 +275,42 @@ describe("Fair", () => {
               hasNextPage
               endCursor
             }
+            pageCursors {
+              around {
+                page
+              }
+              first {
+                page
+              }
+              previous {
+                page
+              }
+            }
           }
         }
       }
     `
 
     const data = await runQuery(query, context)
-    expect(data).toEqual({
+
+    const {
       fair: {
-        shows: {
-          pageInfo: {
-            endCursor: "1234567890",
-            hasNextPage: true,
-          },
-        },
+        shows: { pageCursors, pageInfo },
       },
+    } = data
+
+    const { around, first, previous } = pageCursors
+
+    expect(around.length).toBe(4)
+    expect(first).toBe(null)
+    expect(previous).toBe(null)
+    expect(pageInfo).toEqual({
+      endCursor: null,
```

Comment:
**blocking:** This is a surprise to me - why is it `null`? In order to maintain backwards compatibility, I would expect this to equal `1234567890` since that is what it was in the previous test.

Reply:
I was sure that [this method](https://github.com/artsy/metaphysics/blob/9fd312b2d27b2678d96aa7e5fbf9beedaf9244d2/src/schema/v2/fair.ts#L300) would return `pageInfo` object with proper `endCursor` field. [[connectionFromArraySlice method type]](https://github.com/graphql/graphql-relay-js/blob/b4cb206c13fc901b4e1e4f63299ac8f8658455c8/src/connection/arrayConnection.d.ts#L41)

I should've thought that my changes might be breaking 
And moreover it turned out not to be so easy and I had to dig a bit more into depth. And if I understand it right we invoke [this method](https://github.com/artsy/gravity/blob/fa4ed7b208f9f34129f2f9550c2d736cad3149b5/app/api/v1/fairs_shows_endpoint.rb#L41) on Gravity when paginating and it doesn't support jumping to a certain page, it only supports loading next bunch of results, so it seems like we have to make some changes on Gravity in order to change pagination logic of fair exhibitors on Force

So I was definitely in a hurry to merge this PR 😅  I expected that I would be able to adjust pagination once I have access to Force locally

Please correct me if I'm wrong

Reply:
@anastasiapyzhik all good! We have to be extra careful with breaking changes to Metaphysics because it is also consumed by our iOS and Android apps.

That means that anyone using an older version of the app is making queries using an outdated version of our schema. So if we change the way a field works, it could cause their apps to break.

In this case, I believe it's not a problem because [we don't seem to be using that endCursor field in Eigen](https://github.com/artsy/eigen/blob/74722c329bb74f2851b10092c95df17c3e22557e/src/lib/Scenes/Fair/Components/FairExhibitors.tsx#L83-L113). There is only a "show more" button that fetches the next page, and no "last page" button that would require an end cursor.

It's possible that older versions of the app _did_ use `endCursor`, but hard to know with 100% confidence whether or not that is the case.

One other thing I realized while looking at this and the attached Jira ticket: do we actually need this change? I've started a [thread with Jonathan in Slack ](https://artsy.slack.com/archives/C9SATFLUU/p1625650842112600) to clarify the requirement.

If we do need the change, I think likely OK to merge. Let's see what Jonathan says!

Reply:
@mdole I've already asked Jonathan about pagination [here](https://artsy.slack.com/archives/C9SATFLUU/p1625039727429100) so I was certain that these changes are relevant. Now I see that it should be separated from the current ticket, so I'll focus on sorting for now.

What do you think about pagination supporting by Gravity? I'm not good at ruby but as I see [this method](https://github.com/artsy/gravity/blob/fa4ed7b208f9f34129f2f9550c2d736cad3149b5/app/api/v2/util/cursor_helpers.rb#L13) doesn't process `page` value which will be surely needed

Reply:
Oh awesome, sorry I missed that you had already checked in about pagination! I created a [separate ticket ](https://artsyproduct.atlassian.net/browse/FX-3071) for pagination in Jira.

And thanks for your patience while I looked into your question about pagination in Gravity. Short answer: yes, with the way the `fair_shows_endpoint` is currently set up, it would not allow us to easily paginate using Metaphysics. However, I think this is very fixable. I have included details in the above Jira ticket and assigned it to myself; I will hopefully complete it in the next 1 - 2 days.

Does that sound OK to you? Are you blocked by this pagination problem, or do you have sufficient other work?

Reply:
@mdole that's totally OK, this issue doesn't block me. Thank you for diving into that!